### PR TITLE
Add utils

### PR DESCRIPTION
**Pull request by sohil-kshirsagar**
## Summary

- Added utility modules to the codebase

## Testing

- [ ] Verify utility functions work as expected

### CODE REVIEW

File: utils/string_utils.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def capitalize_first_letter(string):
+    """
+    Capitalize the first letter of a string
+    If string is None, return empty string
+    Otherwise, return the string with the first letter capitalized
+    """
+    return string[0].upper() + string[1:]
```
Comment (ghost) [inline utils/string_utils.py:11]: P2: `capitalize_first_letter()` crashes with `TypeError` when passed `None` because the function lacks a None check despite the docstring promising to return an empty string.

<details>
<summary>AI prompts</summary>

**Check + fix prompt**
```text
Check if this issue is valid. If so, find the root cause and fix it.

<file_path>utils/string_utils.py</file_path>
<line_numbers>7</line_numbers>

<comment>The docstring states 'If string is None, return empty string', but there is no None check. Calling `capitalize_first_letter(None)` will raise `TypeError: 'NoneType' object is not subscriptable` on `string[0]`.</comment>
```

**Check prompt**
```text
Check if this issue is valid. If so, find the root cause and let me know. Don't update any code just yet.

<file_path>utils/string_utils.py</file_path>
<line_numbers>7</line_numbers>

<comment>The docstring states 'If string is None, return empty string', but there is no None check. Calling `capitalize_first_letter(None)` will raise `TypeError: 'NoneType' object is not subscriptable` on `string[0]`.</comment>
```

</details>

---
✅ **Resolved** in commit [8b2fa40](https://github.com/sohil-kshirsagar/test-repo/pull/185/commits/8b2fa409d4e8ea222c262dcf275357509cd416c4)

Comment (ghost) [inline utils/string_utils.py:11]: P2: `capitalize_first_letter("")` raises `IndexError` on `string[0]` due to missing empty-string guard. Add `if not string: return ""` check.

<details>
<summary>AI prompts</summary>

**Check + fix prompt**
```text
Check if this issue is valid. If so, find the root cause and fix it.

<file_path>utils/string_utils.py</file_path>
<line_numbers>7</line_numbers>

<comment>Calling `capitalize_first_letter("")` will raise `IndexError: string index out of range` on `string[0]`, since no empty-string guard exists. The docstring implies the function should handle edge cases gracefully.</comment>
```

**Check prompt**
```text
Check if this issue is valid. If so, find the root cause and let me know. Don't update any code just yet.

<file_path>utils/string_utils.py</file_path>
<line_numbers>7</line_numbers>

<comment>Calling `capitalize_first_letter("")` will raise `IndexError: string index out of range` on `string[0]`, since no empty-string guard exists. The docstring implies the function should handle edge cases gracefully.</comment>
```

</details>

---
✅ **Resolved** in commit [8b2fa40](https://github.com/sohil-kshirsagar/test-repo/pull/185/commits/8b2fa409d4e8ea222c262dcf275357509cd416c4)
